### DISCUSSION TITLE
[build-system] requires does not fail on mismatch

### DISCUSSION THREAD

**Discussion by aethanol** [python-poetry/poetry#3688] [Q&A] [answered]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **macOS**: 
- **1.0.10**: 
- **pyproject.toml**: 
```
...
[build-system]
requires = ["poetry==1.0.10"]
build-backend = "poetry.masonry.api"
```

## Issue

I am attempting to pin to a specific version of poetry in Docker and in the pyproject.toml, but it does not fail when installing while using a mismatching poetry version.
In this case, I've updated the pyproject.toml file to 1.0.10, and am using 1.0.9 locally.

**Comment by abn** [comment, marked as answer]
@aethanol I think you are misunderstanding [PEP 517](https://www.python.org/dev/peps/pep-0517/) here. The purpose for this is it allow build system to determine what build backend to use when building a package. For example if you install a poetry managed managed project using pip or any other PEP 517 compatible build front ends, eg: `pip install /path/to/package` or `pip install source.tar.gz` etc. In these example pip will create a new isolated temporary build environment with the `build-system.requires` present along with the source and call the `build-backend` method to produce a `wheel` or `sdist` it can install to the active environment. Poetry itself does not take this into account for poetry managed projects.

This is not intended to control the version of poetry made available to the user in their enviornment. This is should be managed outside the context of the project.
